hash-parser/src/emit.rs

```diff
@@ -156,8 +156,8 @@ impl ast::IntoAstNode<ast::Type> for HashPair<'_> {
     }
 }
 
-impl ast::IntoAstNode<ast::Literal> for HashPair<'_> {
-    fn into_ast(self) -> ast::AstNode<ast::Literal> {
+impl IntoAstNode<Literal> for HashPair<'_> {
+    fn into_ast(self) -> AstNode<Literal> {
         match self.as_rule() {
             Rule::literal_expr => {
                 let ab = AstBuilder::from_pair(&self);
```

Comment:
Did we decide on whether we should have non-float exponents? My vote would be no.

Reply:
yep i removed this ability